capstone/src/main/java/com/google/sps/servlets/StudySetServlet.java

```diff
@@ -236,8 +239,14 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
       );
       return;
     }
+    String ownerID = "";
+    try {
+      ownerID = getOwnerId(pool);
+    } catch (NotLoggedInException ex) {
+      throw new RuntimeException("User must be logged in", ex);
```

Comment:
Closer, but from my last comment you ideally want to return a 401 error or similar to the user. Replace line 246 with setting the status code and adding this error message to the response body.